packages/pwa/app/components/dynamic-image/index.jsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import PropTypes from 'prop-types'
+import {Img, Box} from '@chakra-ui/react'
+import {getResponsiveImageAttributes} from '../../utils/responsive-image'
+
+/**
+ * Quickly create a responsive image using your dynamic image service
+ * @example
+ *  // Widths without a unit are interpreted as px values
+ * <DynamicImage src="http://example.com/image.jpg[?sw={width}&q=60]" widths={[100, 360, 720]} />
+ * <DynamicImage src="http://example.com/image.jpg[?sw={width}&q=60]" widths={{base: 100, sm: 360, md: 720}} />
+ * // You can also use units of px or vw
+ * <DynamicImage src="http://example.com/image.jpg[?sw={width}&q=60]" widths={['50vw', '100vw', '500px']} />
+ */
+const DynamicImage = ({src, widths, imageProps, as, ...rest}) => {
+    const Component = as ? as : Img
```

Comment:
So I was digging around and I noticed that the utility function used to get the responsive image properties was getting called a lot (on the pdp), as it turns out there is probably something wrong with the page itself because the hero image in the gallery was re-rendering a lot casing this component to mount and unmount a lot.

I initially thought that there was something up with this component and thought that using the useMemo hook to only compute these values once would fix it like this:

```
    const responsiveProps = useMemo(() => getResponsiveImageAttributes({src, widths}), [
        src,
        widths
    ])
```

But because it's actually being mounted and unmounted from it's parent there is nothing you can do. There is no real action here. But I just wanted to show you my though pattern and its usually a good idea to use the useMemo hook for operations like this.

Reply:
Another crazy thought...

Is it possible that because we don't re-calculate the responsive attributes on browser size change, that we get into a weird state where we have pixelated images for a specific window width, or conversely an overly large image for a small window?  

I don't expect you to fix that right now, but if it's the case then maybe we can have a known issue with a follow up.

Reply:
Yes, `useMemo` is still useful for a complex operation like `getResponsiveImageAttributes`. I've now used it.

Reply:
Regarding `we don't re-calculate the responsive attributes on browser size change`, as we've discussed last Friday, this is not a worry because we calculate based on the theme's breakpoints (unlike window's width, they're available on the server-side).